lib/python/qmk/cli/generate/version_h.py

```diff
@@ -34,25 +34,37 @@ def generate_version_h(cli):
         git_qmk_hash = "NA"
         chibios_version = "NA"
         chibios_contrib_version = "NA"
+        # these variable aren't wrapped on quotes when generating the version file as they are number when git isn't skipped
+        # we add double quoting here such that generated code is valid
+        qmk_major, qmk_minor, qmk_patch, qmk_global = ['"NA"'] * 4
     else:
         git_dirty = git_is_dirty()
         git_version = git_get_version() or current_time
         git_qmk_hash = git_get_qmk_hash() or "Unknown"
         chibios_version = git_get_version("chibios", "os") or current_time
         chibios_contrib_version = git_get_version("chibios-contrib", "os") or current_time
+        qmk_major, qmk_minor, qmk_patch = git_get_qmk_major_minor_patch()
+        # number that can be used to order all versions, eg '#if __QMK__ == 000020005' (for 0.20.5)
+        # TODO: proper handling when function fails and returns "NA", so int() doesn't error out (?)
+        # FIXME: may need to reserve more than 3 digits for each number
+        qmk_global = int(f"{qmk_major:03}{qmk_minor:03}{qmk_patch:03}")
+
+    dirty_indicator = "*" if git_dirty else ""
 
     # Build the version.h file.
     version_h_lines = [GPL2_HEADER_C_LIKE, GENERATED_HEADER_C_LIKE, '#pragma once']
 
-    version_h_lines.append(
-        f"""
-#define QMK_VERSION "{git_version}"
-#define QMK_BUILDDATE "{current_time}"
-#define QMK_GIT_HASH  "{git_qmk_hash}{'*' if git_dirty else ''}"
-#define CHIBIOS_VERSION "{chibios_version}"
-#define CHIBIOS_CONTRIB_VERSION "{chibios_contrib_version}"
-"""
-    )
+    version_h_lines.extend([
+        f'#define QMK_VERSION "{git_version}"',
+        f'#define QMK_MAJOR {qmk_major}',
+        f'#define QMK_MINOR {qmk_minor}',
+        f'#define QMK_PATCH {qmk_patch}',
+        f'#define __QMK__ {qmk_global}',
```

Comment:
```suggestion
        f'#define __QMK__ 0x{qmk_major:02}{qmk_minor:02}{qmk_patch:04}ul', # number that can be used to order all versions, eg '#if __QMK__ == 0x00200005ul' (for 0.20.5) -- prefer to have a 0x prefix otherwise leading zero's would be interpreted as octal numbers.
```